bin/compile

```diff
@@ -18,7 +35,17 @@ cd $BUILD_DIR
 if [ -f shard.yml ]; then
   echo "-----> Installing Dependencies"
   $CRYSTAL_DIR/bin/crystal deps
+
+  if [ -f app.cr ]; then
+    echo "-----> Compiling app.cr (using old behaviour, app.cr exists - ignoring shard.yml)"
+    $CRYSTAL_DIR/bin/crystal build app.cr --release
+  else
+    eval $(parse_yaml shard.yml "shard_")
+    echo "-----> Compiling src/${shard_name}.cr (auto-detected from shard.yml)"
+    $CRYSTAL_DIR/bin/crystal build src/${shard_name}.cr --release -o app
+  fi
+else
+  echo "-----> Compiling app.cr (create shard.yml to override, learn more: https://github.com/ysbaddaden/shards)"
```

Comment:
I would issue a deprecation here as well, but with a message like `DEPRECATED: Your using an old version of the dependency manager, please look at https://github.com/ysbaddaden/shards for information on the currently advised method\nCompiling app.cr`

I don't think `echo` supports newlines without some flags, but you might be able to get around it with a `printf`.